src/objective-c/GRPCClient/private/GRPCCore/GRPCSecureChannelFactory.m

```diff
@@ -60,8 +61,13 @@ - (instancetype)initWithPEMRootCerts:(NSString *)rootCerts
     NSBundle *bundle = [NSBundle bundleForClass:[self class]];
     NSBundle *resourceBundle = [NSBundle
         bundleWithURL:[[bundle resourceURL] URLByAppendingPathComponent:resourceBundlePath]];
-    NSString *path = [resourceBundle pathForResource:rootsPEM ofType:@"pem"];
-    setenv("GRPC_DEFAULT_SSL_ROOTS_FILE_PATH", [path cStringUsingEncoding:NSUTF8StringEncoding], 1);
+    _sslRootPathStr = [resourceBundle pathForResource:rootsPEM ofType:@"pem"];
+    const char *utf8Str = [_sslRootPathStr cStringUsingEncoding:NSUTF8StringEncoding];
+    if (utf8Str != NULL) {
+      setenv("GRPC_DEFAULT_SSL_ROOTS_FILE_PATH", utf8Str, 1);
+    } else {
+      NSLog(@"Failed to set default ssl root file path.");
```

Comment:
Is a log good enough here? Is it appropriate to return an error here? Not familiar with how this should work.

Reply:
thanks. we actually use a mixture of NSLog and NSError in the code base at the moment; Looks like here we've already setup the NSError reporting path, so gonna updated it to report via the error object instead (which is actually a better choice I think ) ; )